python_modules/libraries/dagster-databricks/dagster_databricks/configs.py

```diff
@@ -625,6 +650,115 @@ def _define_libraries() -> Field:
+    return {
+        "no_alert_for_canceled_runs": no_alert_for_canceled_runs,
+        "no_alert_for_skipped_runs": no_alert_for_skipped_runs,
+    }
+
+
+def _define_webhook_notification_settings() -> Field:
+    webhook_id_field = Field(
+        [
+            Shape(
+                {
+                    "id": Field(String, is_required=False),
+                }
+            )
+        ],
+        is_required=False,
+    )
+    return Field(
+        Shape(
+            {
+                "on_duration_warning_threshold_exceeded": webhook_id_field,
+                "on_failure": webhook_id_field,
+                "on_start": webhook_id_field,
+                "on_success": webhook_id_field,
+            }
+        ),
+        is_required=False,
+        description="Optional webhooks to trigger at different stages of job execution",
+    )
+
+
+def _define_docker_image_conf() -> Field:
+    docker_basic_auth = Shape(
+        {
+            "password": Field(String),
+            "username": Field(String),
+        }
+    )
+    basic_auth = Field(
+        docker_basic_auth,
+        description=(
+            "Authentication information for pulling down docker image. Leave unconfigured if the"
+            " image is stored in AWS ECR you have an instance profile configured which already has"
+            " permissions to pull the image from the configured URL"
+        ),
+        is_required=False,
+    )
+    url = Field(str, description="Image URL")
+    return Field(
+        Shape({"basic_auth": basic_auth, "url": url}),
+        description="Optional Docker image to use as base image for the cluster",
+        is_required=False,
+    )
+
+
+def _define_job_health_settings():
+    jobs_health_rule = Shape(
+        {
+            "metric": Field(
+                Enum("JobsHealthMetric", [EnumValue("RUN_DURATION_SECONDS")]),
+                is_required=True,
+                description=(
+                    "Specifies the health metric that is being evaluated for a particular health"
+                    " rule"
+                ),
+            ),
+            "op": Field(
+                Enum("JobsHealthOperator", [EnumValue("GREATER_THAN")]),
+                is_required=True,
+                description=(
+                    "Specifies the operator used to compare the health metric value with the"
+                    " specified threshold"
+                ),
+            ),
+            "value": Field(
+                Int,
+                is_required=True,
+                description=(
+                    "Specifies the threshold value that the health metric should obey to satisfy"
+                    " the health rule."
+                ),
+            ),
+        }
+    )
+    return Field(
+        [jobs_health_rule],
+        is_required=False,
+        description="An optional set of health rules that can be defined for this job",
+    )
+
+
 def _define_submit_run_fields() -> Dict[str, Union[Selector, Field]]:
```

Comment:
re: pyright, this type is no longer accurate (it's now a `Dict[str, Union[Selector, Field, Dict[str, Union[Selector, Field]]]]` 😵 )

no big deal if you just replace this with `Dict[str, Any]`.